src/camera.rs

```diff
@@ -56,18 +60,43 @@ impl Camera {
         }
     }
 
+    pub fn on_mouse(&mut self, delta: (f64, f64)){
+        let view_rot_speed = cgmath::Rad(0.01f32);
+        let dx = delta.0 as f32;
+        let dy = delta.1 as f32;
+        self.rotation.0 = self.rotation.0 - view_rot_speed * dx;
+        self.rotation.1 = self.rotation.1 - view_rot_speed * dy;
+
+        // clamping
```

Comment:
Could you move the clamping into a private function which will be called by `on_mouse` and `update`?